File: api-android/src/main/java/com/engarnet/timetree/model/TActivity.kt
package com.engarnet.timetree.model

import com.engarnet.timetree.api.v1.api.activities.response.ActivityResponse
import java.util.*

data class TActivity(
    val id: String,
    val content: String,
    val updatedAt: Date,
    val createdAt: Date
)

internal fun ActivityResponse.toModel(): TActivity {
    return data.let {
        TActivity(
            id = it.id,
            content = it.attributes.content,
            updatedAt = Date().apply { time = it.attributes.updatedAt.time },
            createdAt = Date().apply { time = it.attributes.createdAt.time }
        )
    }
}

File: api-ios/TimeTreeAPI/Sources/Model/TActivity.swift
//
//  TActivity.swift
//  TimeTreeAPI
//
//  Created by EngarNet on 2020/05/02.
//  Copyright © 2020 EngarNet. All rights reserved.
//

import Foundation
import TimeTreeAPICommon

public class TActivity {
    public init(
        id: String,
        content: String,
        updatedAt: Foundation.Date,
        createdAt: Foundation.Date
    ) {
        self.id = id
        self.content = content
        self.updatedAt = updatedAt
        self.createdAt = createdAt
    }

    public let id: String
    public let content: String
    public let updatedAt: Foundation.Date
    public let createdAt: Foundation.Date
}

extension ActivityResponse {
    func toModel() -> TActivity {
        return TActivity(
            id: self.data.id,
            content: self.data.attributes.content,
            updatedAt: Foundation.Date(timeIntervalSince1970: TimeInterval(self.data.attributes.updatedAt.time)),
            createdAt: Foundation.Date(timeIntervalSince1970: TimeInterval(self.data.attributes.createdAt.time))
        )
    }
}
